OpenRA.Mods.Common/Traits/Power/Player/PowerManager.cs

```diff
@@ -128,7 +129,7 @@ void UpdatePowerState()
 				UpdatePowerRequiringActors();
 
 			if (isLowPower && !wasLowPower)
-				nextPowerAdviceTime = 0;
+				lastPowerAdviceTime = Game.RunTime + info.AdviceInterval;
```

Comment:
The original code set `nextPowerAdviceTime = 0` to guarantee that the notification plays in `Tick`. Setting `lastPowerAdviceTime = Game.RunTime + info.AdviceInterval` guarantees that it *won't* play when the player triggers low power!

The simplest fix here would be to set `lastPowerAdviceTime = -info.AdviceInterval` with a comment "Force the notification to play immediately".

We'll also need to initialize this in the ctor like the rest.

Reply:
> We'll also need to initialize this in the ctor like the rest.

Actually, probably not in this case: this would cause the "low power" to play on the first tick on scripted maps where you start in low power, which would be bad. The code here makes sure that the notification will always play when the player triggers low power, which is good enough.